packages/truffle/test/scenarios/compile/compile.js

```diff
@@ -385,3 +385,38 @@ describe("Repeated compilation of contracts with inheritance [ @standalone ]", f
     }
   });
 });
+
+describe("Compilation with db enabled", async () => {
+  const logger = new MemoryLogger();
+
+  function checkForDb(config) {
+    const dbPath = path.join(config.working_directory, ".db");
+    const dbExists = fse.pathExistsSync(dbPath);
+    return dbExists;
+  }
+
+  before("set up the server", function (done) {
+    Server.start(done);
+  });
+
+  after("stop server", function (done) {
+    Server.stop(done);
+  });
+
+  it("creates a .db directory when db is enabled", async function () {
+    this.timeout(12000);
+    const project = path.join(__dirname, "../../sources/contract_names");
+    const config = await sandbox.create(project);
+
+    try {
+      await CommandRunner.run("compile", config);
+    } catch (error) {
+      output = logger.contents();
+      log(output);
+      throw new Error(error);
+    }
+
+    const dbExists = checkForDb(config);
+    assert(dbExists === true);
```

Comment:
This should also test that a contract made it into the db, no?

Reply:
Added! I just test that the contract names exist in the db, since I think that getting deeper into the contents of the db should be a task for the db tests, right?

Reply:
Yeah, agreed!